### DISCUSSION TITLE
DNS resolver while installing TALOS on Proxmox

### DISCUSSION THREAD

**Discussion by glycerin-ce** [siderolabs/talos#6389] [General]
Hi everybody.
I have to install 2 Talos VM on a  proxmox but I cannot use the Google resolver (8.8.8.8).
Is there a hint on how to use another resolver?
Cause the network must use an internal one.
![Screenshot from 2022-10-10 18-45-14](https://user-images.githubusercontent.com/82819640/194915473-4b21f016-c6ba-4168-8848-e883bcd054a6.png)

**Comment by smira** [comment]
There are multiple ways to set DNS resolver, but the actual answer depends on your setup:

* DHCP might set the resolvers
* Talos machine config can set the resolvers as `.machine.network.nameservers`
* Talos platform (`nocloud`) can provide the resolvers
* kernel command line can provide the resolvers via `ip=` kernel argument

When running on Proxmox via `nocloud` Talos platform, setting it via the Proxmox is the best way I think.

**Reply by glycerin-ce** [reply]
Thanks @smira for your suggestion.
Now something similar appear at boot time.
The Talos ISO ask try to get in touch with soem ntp server while these are forbidden inside this environment.
Do you already know how to set these at boot time?
I hope so like the one used for setting the DNS server.

**Reply by smira** [reply]
NTP servers are set pretty much same way as DNS servers.

**Reply by glycerin-ce** [reply]
Found it
`ip=<client-ip>::<gw_ip>:<netmask>::<device>:<autoconof>:<DNS_ip>::<NTP_ip>`

but now the installation process seems to be in stuck after adjusting time with my own NP server.
Do I forgot other info while setting the Ip address at boot time?
![Screenshot from 2022-11-07 12-51-51](https://user-images.githubusercontent.com/82819640/200303535-750af07e-ef37-42c1-aee9-7f37a303864b.png)

**Reply by smira** [reply]
Installation process is not stuck, Talos is waiting for machine config to be applied.

https://www.talos.dev/v1.2/introduction/getting-started/#apply-configuration
